rest/server_context.go

```diff
@@ -42,14 +41,38 @@ const DefaultStatsLogFrequencySecs = 60
 // This struct is accessed from HTTP handlers running on multiple goroutines, so it needs to
 // be thread-safe.
 type ServerContext struct {
-	config            *ServerConfig
-	databases_        map[string]*db.DatabaseContext
-	lock              sync.RWMutex
-	statsContext      *statsContext
-	HTTPClient        *http.Client
-	replicator        *base.Replicator
-	cpuPprofFileMutex sync.Mutex // Protect cpuPprofFile from concurrent Start and Stop CPU profiling requests
-	cpuPprofFile      *os.File   // An open file descriptor holds the reference during CPU profiling
+	config           *StartupConfig
+	persistentConfig bool
+	// bucketDbConfigs is a map of bucket name to DatabaseConfig
+	bucketDbConfigs map[string]*DatabaseConfig
+	// dbConfigs is a map of db name to DatabaseConfig
+	dbConfigs map[string]*DatabaseConfig
+	// databases_ is a map of dbname to db.DatabaseContext
+	databases_          map[string]*db.DatabaseContext
+	lock                sync.RWMutex
+	legacyReplications  []*ReplicateV1Config
+	statsContext        *statsContext
+	bootstrapConnection *gocb.Cluster
+	HTTPClient          *http.Client
+	replicator          *base.Replicator
+	cpuPprofFileMutex   sync.Mutex // Protect cpuPprofFile from concurrent Start and Stop CPU profiling requests
+	cpuPprofFile        *os.File   // An open file descriptor holds the reference during CPU profiling
+}
+
+type DatabaseConfig struct {
+	CAS    gocb.Cas
```

Comment:
There's no usage of CAS yet, but would be interested in the details of how it's expected to be used (or omit it until it's implemented).

Reply:
Will remove for now, but will be used for an early-exit opportunity for config updates